core/src/apps/cardano/sign_tx/ordinary_signer.py

```diff
@@ -25,6 +25,8 @@ def __init__(
     ) -> None:
         super().__init__(ctx, msg, keychain)
 
+        self.signing_mode_title = "Confirming a transaction."
```

Comment:
maybe make this a classvar?

Reply:
Like this https://github.com/trezor/trezor-firmware/pull/2355/commits/c702e9db85293bc10b4f87bc1e01aca4d8e217d7? I didn't find a way to mark it abstract so that we could get rid of the assert 🤔

Reply:
exactly.
I would expect `ClassVar[str]` without specifying the value would be valid?
(but maybe the typechecker would only allow that if the class was abstract via abcmeta, which we probably can't do in uPy)
but if not then let's keep it like this for now

Reply:
It is indeed valid. But the typechecker doesn't enforce the value to be set in the child signers so I'm not sure what the best approach here is.

I'd prefer keeping the type `ClassVar[str | None]` so that the assert in `_show_tx_init` is required and the app crashes "predictably" if `SIGNING_MODE_TITLE` is not set in any of the child classes. Also I would add a comment to the assert and perhaps also to the variable declaration in `Signer` if we decide to keep it.

Reply:
I filed a bug https://github.com/microsoft/pyright/issues/3652, but I'm not sure if it can be fixed in a practical manner.

I don't think that the assert is bringing any value except appeasing the type checker though. You are still asserting immediately before using the value, so the difference is only between getting an `AssertionError` in debug mode, or `AttributeError` in both debug and production. Using a semantically wrong type annotation does not help us do what we want, i.e., check this statically; and good test suite coverage can catch either option.

So please change to `ClassVar[str]` and remove the assert.

Reply:
Done in https://github.com/trezor/trezor-firmware/pull/2355/commits/3babd23acb7a4029e9e51b16fe734a194c1c8630.